### ISSUE TITLE
Lint/NestedMethodDefinition false alarm on the `@` presence

### ISSUE THREAD

**Issue by yegor256** [rubocop/rubocop#12960]
I'm doing this:

```
class F
  def foo
    @x = Object.new
    def @x.bar(_t)
      true
    end
  end
end
```

Rubocop 1.64.0 complains:

```
a.rb:4:5: W: Lint/NestedMethodDefinition: Method definitions must not be nested. Use lambda instead.
    def @x.bar(_t) ...
    ^^^^^^^^^^^^^^
```

Looks like a bug.

More details here: https://stackoverflow.com/questions/78573984/how-to-define-a-ruby-method-in-runtime-so-that-rubocop-doesnt-complain?noredirect=1#comment138524046_78573984

**Comment by yegor256** [comment]
@koic thanks!
